src/devices/honeywell_cm921.c

```diff
@@ -0,0 +1,449 @@
+    const int row = 0; // we expect a single row only.
+
+    if (bitbuffer->num_rows != 1 || bitbuffer->bits_per_row[row] < 60)
+        return DECODE_ABORT_LENGTH;
+
+    if (decoder->verbose)
+        bitrow_printf(bitbuffer->bb[row], bitbuffer->bits_per_row[row], "%s: ", __func__);
+
+    int preamble_start = bitbuffer_search(bitbuffer, row, 0, preamble_pattern, preamble_bit_length);
+    int start = preamble_start + preamble_bit_length;
+    int len = bitbuffer->bits_per_row[row] - start;
+    if (decoder->verbose)
+        printf("preamble_start=%d start=%d len=%d\n", preamble_start, start, len);
+    if (len < 8) return DECODE_ABORT_LENGTH;
+    int end = start + len;
+
+    bitbuffer_t bytes = {0};
+    int pos = start;
+    while (pos < end) {
+        uint8_t byte = 0;
+        if (get_byte(bitbuffer, row, pos, end, &byte) != 10)
+            break;
+        for (unsigned i = 0; i < 8; i++)
+            bitbuffer_add_bit(&bytes, (byte >> i) & 0x1);
+        pos += 10;
+    }
+
+    // Skip Manchester breaking header
+    uint8_t header[3] = { 0x33, 0x55, 0x53 };
+    if (bitrow_get_byte(bytes.bb[row], 0) != header[0] ||
+        bitrow_get_byte(bytes.bb[row], 8) != header[1] ||
+        bitrow_get_byte(bytes.bb[row], 16) != header[2])
+        return DECODE_FAIL_SANITY;
+
+    // Find Footer 0x35 (0x55*)
+    int fi = bytes.bits_per_row[row] - 8;
+    int seen_aa = 0;
+    while (bitrow_get_byte(bytes.bb[row], fi) == 0x55) {
+        seen_aa = 1;
+        fi -= 8;
+    }
+    if (!seen_aa || bitrow_get_byte(bytes.bb[row], fi) != 0x35)
+        return DECODE_FAIL_SANITY;
+
+    unsigned first_byte = 24;
+    unsigned end_byte = fi;
+    unsigned num_bits  = end_byte - first_byte;
+    unsigned num_bytes = num_bits/8 / 2;
+
+    bitbuffer_t packet = {0};
+    unsigned fpos = bitbuffer_manchester_decode(&bytes, row, first_byte, &packet, num_bits);
+    unsigned man_errors = num_bits  - (fpos - first_byte - 2);
+    char *packet_hex = hex(&packet, row);
+
+#ifndef _DEBUG
+    if (man_errors != 0)
+        return DECODE_FAIL_SANITY;
+#endif
+
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",        "", DATA_STRING, "Honeywell CM921",
+            "Packet",       "", DATA_STRING, packet_hex,
+            NULL);
+    /* clang-format on */
+
+    message_t message;
+
+    if (parse_msg(&packet, row, &message))
+        data = interpret_message(&message, data);
+
+#ifdef _DEBUG
+    data = add_hex_string(data, "Header", &message.header, 1);
+    uint8_t cmd[2] = { message.command >> 8, message.command & 0x00FF};
+    data = add_hex_string(data, "Command", cmd, 2);
+    data = add_hex_string(data, "Payload", message.payload, message.payload_length);
+    data = add_hex_string(data, "Unparsed", message.unparsed, message.unparsed_length);
+    data = add_hex_string(data, "CRC", &message.crc, 1);
+    data = data_append(data, "# man errors", "", DATA_INT, man_errors, NULL);
+#endif
+
```

Comment:
Here somewhere you should add that the message has a MIC of type CHECKSUM. Look in other decoders for reference.

Reply:
Done!